lib/ctf-ir/fields.c

```diff
@@ -1859,16 +1787,33 @@ struct bt_ctf_field *bt_ctf_field_structure_create(
 		goto end;
 	}
 
-	structure->field_name_to_index = structure_type->field_name_to_index;
 	structure->fields = g_ptr_array_new_with_free_func(
-		(GDestroyNotify)bt_ctf_field_put);
+		(GDestroyNotify) bt_ctf_field_put);
 	g_ptr_array_set_size(structure->fields,
-		g_hash_table_size(structure->field_name_to_index));
-	field = &structure->parent;
-	BT_LOGD("Created structure field object: addr=%p, ft-addr=%p",
-		field, type);
+		structure_type->fields->len);
+
+	/* Create all fields contained by the structure field. */
+	for (i = 0; i < structure_type->fields->len; i++) {
+		struct bt_ctf_field *field;
+		struct structure_field *field_type =
+			g_ptr_array_index(structure_type->fields, i);
+
+		field = bt_ctf_field_create(field_type->type);
+		if (!field) {
+			BT_LOGE("Failed to create structure member=\'%s\'",
```

Comment:
**BAD** log message, use:

    "Failed to create structure field's member: name=\"%s\", index=%zu"